testing/test_pdb.py

```diff
@@ -1508,3 +1508,40 @@ def inner():
 # inner()
 # c
 """)
+
+
+def test_set_trace_in_completion():
+    def fn():
+        class CompleteMe(object):
+            attr_called = 0
+
+            @property
+            def set_trace_in_attrib(self):
+                self.attr_called += 1
+                set_trace(cleanup=False)
+                print("inner_set_trace_was_ignored")
+
+        obj = CompleteMe()
+
+        set_trace()
+
+        comps = []
+        while True:
+            val = pdb.GLOBAL_PDB.complete("obj.", len(comps))
+            if val is None:
+                break
+            comps += [val]
+        assert obj.attr_called == 1, "attr was called"
+        # XXX: on py27/pypy on Travis this contains color codes?!
+        # (e.g. '\x1b[000;00m\x1b[35;01m__class__\x1b[00m')
+        # (But works locally for me using "tox -e py27")
+        if sys.version_info > (3, ):
+            assert "set_trace_in_attrib" in comps
```

Comment:
@antocuni 
Any idea why that is?  (colors for py27/pypy here, but not the others)
There is some other test that gets skipped unconditionally because of escapes IIRC.

Reply:
The difference is if `pyrepl.readline` can be used, or not.
Fails for me due to https://bitbucket.org/pypy/pyrepl/issues/19/does-not-find-curses-via-ctypes-on-arch.

Reply:
(fixed/addressed in the test)